ihatemoney/web.py

```diff
@@ -391,6 +396,83 @@ def edit_project():
     )
 
 
+@main.route("/<project_id>/upload_json", methods=["GET", "POST"])
+def upload_json():
+    form = UploadForm()
+    pid = g.project.id
+    if form.validate_on_submit():
+        filename = pid + "_uploaded_bills.json"
+        form.file.data.save(filename)
+        try:
+            import_project(filename)
+            flash(_("Project successfully uploaded"))
+        except ValueError:
+            flash(_("Invalid JSON"), category="error")
+        finally:
+            os.remove(filename)
+        return redirect(url_for("main.list_bills"))
+
+    return render_template("upload_json.html", form=form)
+
+
+def import_project(file):
+    # From json : export list of members
+    json_file = json.load(open(file))
```

Comment:
if we use a file descriptor rather than a filename here we would fix the previous issue.